### PR TITLE
Np 9076 create user creates user

### PR DESCRIPTION
**Pull request by axthosarouris**

### CODE REVIEW

File: cognito-pre-token-generation-openid/src/main/java/no/unit/nva/cognito/UserSelectionUponLoginHandler.java [hunk 1]
```diff
@@ -119,19 +113,6 @@ private static String extractFeideIdentifier(Map<String, String> userAttributes)
         return Optional.ofNullable(userAttributes.get(FEIDE_ID)).orElse(null);
     }
 
-    @JacocoGenerated
```
Comment (axthosarouris) [inline cognito-pre-token-generation-openid/src/main/java/no/unit/nva/cognito/UserSelectionUponLoginHandler.java:122]: Moved to CristinClient

File: user-creation-testing/build.gradle [hunk 2]
```diff
@@ -0,0 +1,53 @@
+plugins {
```
Comment (axthosarouris) [inline user-creation-testing/build.gradle:1]: module for reusing the test setup for the login/user-creation process

File: cognito-pre-token-generation-openid/src/main/java/no/unit/nva/cognito/UserSelectionUponLoginHandler.java [hunk 3]
```diff
@@ -36,27 +30,29 @@
...
 import software.amazon.awssdk.services.cognitoidentityprovider.CognitoIdentityProviderClient;
 import software.amazon.awssdk.services.cognitoidentityprovider.model.AdminUpdateUserAttributesRequest;
 import software.amazon.awssdk.services.cognitoidentityprovider.model.AttributeType;
 
 public class UserSelectionUponLoginHandler
     implements RequestHandler<CognitoUserPoolPreTokenGenerationEvent, CognitoUserPoolPreTokenGenerationEvent> {
 
+    public static final Environment ENVIRONMENT = new Environment();
+
+    public static final Region AWS_REGION = Region.of(ENVIRONMENT.readEnv("AWS_REGION"));
```
Comment (axthosarouris) [inline cognito-pre-token-generation-openid/src/main/java/no/unit/nva/cognito/UserSelectionUponLoginHandler.java:55]: This variable is used only in this handler. Moved from `NetworkingUtils` (deleted class) here.

File: cognito-pre-token-generation-openid/src/main/java/no/unit/nva/cognito/UserSelectionUponLoginHandler.java [hunk 4]
```diff
@@ -70,18 +66,16 @@
 
     @JacocoGenerated
     public UserSelectionUponLoginHandler() {
-        this(defaultCognitoClient(), defaultAuthorizedBackedClient(), CRISTIN_HOST,
+        this(defaultCognitoClient(), CristinClient.defaultClient(),
```
Comment (axthosarouris) [inline cognito-pre-token-generation-openid/src/main/java/no/unit/nva/cognito/UserSelectionUponLoginHandler.java:69]: moved the logic of creating a new default CristinClient to CristinClient.

File: template.yaml [hunk 5]
```diff
@@ -161,7 +162,7 @@ Resources:
                   - cognito-idp:DescribeUserPoolDomain
                 Resource: "*"
 
-  TriggerExecutionRole:
+  CreateUsersRole:
```
Comment (axthosarouris) [inline template.yaml:165]: We do not use the term "trigger" anywhere in the code. The Role is now required by two functions, the trigger and the CreateUser REST API call

File: user-access-handlers/src/main/java/no/unit/nva/handlers/CreateUserHandler.java [hunk 6]
```diff
@@ -1,49 +1,100 @@
...
 
     @Override
     protected UserDto processInput(CreateUserRequest input, RequestInfo requestInfo, Context context)
-        throws ForbiddenException {
+        throws ForbiddenException, NotFoundException {
         authorize(requestInfo);
-        return UserDto.newBuilder()
-            .withInstitution(input.getCustomerId())
-            .build();
+        logger.info("INput:" + attempt(() -> JsonConfig.writeValueAsString(input)).orElseThrow());
```
Comment (axthosarouris) [inline user-access-handlers/src/main/java/no/unit/nva/handlers/CreateUserHandler.java:50]: for debugging reasons. Will be removed in next PR

File: user-access-handlers/src/main/java/no/unit/nva/handlers/CreateUserHandler.java [hunk 7]
```diff
@@ -1,49 +1,100 @@
...
+    }
+
     private void authorize(RequestInfo requestInfo) throws ForbiddenException {
         if (userIsNotAuthorized(requestInfo)) {
             throw new ForbiddenException();
         }
     }
 
     private boolean userIsNotAuthorized(RequestInfo requestInfo) {
+        var loggerInfoString = attempt(() -> JsonConfig.writeValueAsString(requestInfo)).orElseThrow();
```
Comment (axthosarouris) [inline user-access-handlers/src/main/java/no/unit/nva/handlers/CreateUserHandler.java:93]: will be removed in next PR

File: cognito-pre-token-generation-openid/src/test/java/no/unit/nva/cognito/UserSelectionUponLoginHandlerTest.java [hunk 8]
```diff
@@ -105,15 +106,15 @@ public void init() {
...
 
         var authorizedBackedClient =
-            AuthorizedBackendClient.prepareWithBearerToken(HTTP_CLIENT, "Bearer " + authServerMock.getJwtToken());
+            prepareWithBearerToken(HTTP_CLIENT, "Bearer " + authServerMock.getJwtToken());
 
         registeredPeople = new RegisteredPeopleInstance(httpServer, authServerMock, customerService, identityService);
         nvaDataGenerator = new NvaDataGenerator(registeredPeople, customerService);
 
         var cristinHost = this.serverUri;
+        var cristinClient = new CristinClient(cristinHost, authorizedBackedClient);
```
Comment (brinxmat) [inline cognito-pre-token-generation-openid/src/test/java/no/unit/nva/cognito/UserSelectionUponLoginHandlerTest.java:115]: Can you correct the typo in "Backed"

**Review by brinxmat — APPROVED** [review]

**Pull request merged**